### PR TITLE
UNR-2019: Example Project CI

### PR DESCRIPTION
**Pull request by ieva-improbable**
**Contributions**: We are not currently taking public contributions - see our [contributions](CONTRIBUTING.md) policy. However, we are accepting issues and we do want your [feedback](../README.md#give-us-feedback).

-------

#### Description
This PR contains changes required to Unreal GDK to enable Example Project CI. Namely:
* Example Project CI reuses `get-engine.ps1` to obtain Unreal Engine artifact, changes had to be made to pass in location of Engine cache and Engine symlink;
* Example Project builds reuse Engine cache used by Unreal GDK builds, therefore cache location had to be changed to be outside of build folder;
* remove any trailing whitespaces or new lines from the end of string read in from `unreal-engine.version` to avoid accidental build errors.

Example Project PR: https://github.com/spatialos/UnrealGDKExampleProject/pull/57

Engine PR: https://github.com/improbableio/UnrealEngine/pull/254

#### Release note
No release notes, these changes only affect our CI.

#### Tests
No testing by QA needed as these changes only affect our CI. I tested the changes by running builds using this branch.

#### Documentation
Documentation on our CI setup will be added to wiki.

#### Primary reviewers
@m-samiec @ca-johnson

### CODE REVIEW

File: ci/get-engine.ps1 [hunk 1]
```diff
@@ -1,3 +1,10 @@
+param(
+  [string] $unreal_path = "$($gdk_home)\UnrealEngine",
+  # Note: this directory is soutside the build directory and will not get automatically cleaned up from agents unless agents are restarted.
```
Comment (m-samiec) [inline ci/get-engine.ps1]:
```suggestion
  # Note: this directory is outside the build directory and will not get automatically cleaned up from agents unless agents are restarted.
```

**Comment by m-samiec** [comment]
Could you link the engine PR also?

File: ci/get-engine.ps1 [hunk 2]
```diff
@@ -1,3 +1,10 @@
+param(
+  [string] $unreal_path = "$($gdk_home)\UnrealEngine",
+  # Note: this directory is outside the build directory and will not get automatically cleaned up from agents unless agents are restarted.
+  [string] $engine_cache_path = "$gdk_home\..\..\..",
```
[the remark this answers is not in the document]
Comment (ieva-improbable) [reply ci/get-engine.ps1]: Changed it to not use relative path but instead create it directly in the root of the drive. Lmk what you think.

File: ci/setup-gdk.ps1 [hunk 3]
```diff
@@ -7,6 +7,7 @@ pushd "$($gdk_home)"
     $worker_sdk_dir = "$($gdk_home)\SpatialGDK\Source\SpatialGDK\Public\WorkerSDK"
     $worker_sdk_dir_old = "$($gdk_home)\SpatialGDK\Source\Public\WorkerSdk"
     $binaries_dir = "$($gdk_home)\SpatialGDK\Binaries\ThirdParty\Improbable"
+    $msbuild_exe = "${env:ProgramFiles(x86)}\Microsoft Visual Studio\2017\BuildTools\MSBuild\15.0\Bin\MSBuild.exe"
```
Comment (m-samiec) [inline ci/setup-gdk.ps1]: ❤️

File: ci/get-engine.ps1 [hunk 4]
```diff
@@ -1,3 +1,9 @@
+param(
+  [string] $unreal_path = "$($gdk_home)\UnrealEngine",
```
Comment (m-samiec) [inline ci/get-engine.ps1]: nit: Could you add a comment up here adding that this is a symlinked directory into a specific engine version within the $engine_cache_directory?

Comment (ieva-improbable) [reply ci/get-engine.ps1]: Done.

**Pull request merged**